### PR TITLE
OCLOMRS-153: A user should be able to select “Other Languages” when creating a new dictionary

### PR DESCRIPTION
**Pull request by emmabaye**
# JIRA TICKET NAME:
[A user should be able to select “Other Languages” when creating a new dictionary](https://issues.openmrs.org/browse/OCLOMRS-153)

# Summary:
Currently, user can only select preferred language for default locale. User should be able "other languages" for supported locales.

### CODE REVIEW

File: src/components/dashboard/components/dictionary/common/DictionaryModal.jsx [hunk 1]
```diff
@@ -43,6 +46,19 @@ export class DictionaryModal extends React.Component {
       },
     });
   };
+
+  onChangeSupportedLocale = (arr) => {
```
Comment (dondrzzy) [inline src/components/dashboard/components/dictionary/common/DictionaryModal.jsx]: Hey @emmabaye , please name this function name the right way. Like a `do action` such as `changeSupportLocale`. `onChangeSupportLocale` is preferable for an event.

File: src/components/dashboard/components/dictionary/common/DictionaryModal.jsx [hunk 2]
```diff
@@ -82,9 +99,30 @@ export class DictionaryModal extends React.Component {
...
+   const {
+     dictionary: {
+       supported_locales,
+     },
+   } = this.props;
+   if (supported_locales === undefined) {
+     return null;
+   }
+   const supportedLocalesOptions = [];
+   for (let i = 0; i < supported_locales.length; i += 1) {
```
Comment (dondrzzy) [inline src/components/dashboard/components/dictionary/common/DictionaryModal.jsx]: You could have used the short form i++ but that is also okay.

Comment (emmabaye) [reply src/components/dashboard/components/dictionary/common/DictionaryModal.jsx]: Thanks for the feedback. However eslint flags it if I use i++.

File: src/components/dashboard/components/dictionary/common/DictionaryModal.jsx (hunk 1 above)
Comment (emmabaye) [reply src/components/dashboard/components/dictionary/common/DictionaryModal.jsx]: Thanks for the feedback. However,  that is the function that is called when an `option` in the `select` component changes. It is called when an `onChange` event occurs as you can see [here on line 231](https://github.com/openmrs/openmrs-ocl-client/blob/42d13b140e5cdac842ff87de97f244f200392f30/src/components/dashboard/components/dictionary/common/DictionaryModal.jsx).

Comment (emmabaye) [reply src/components/dashboard/components/dictionary/common/DictionaryModal.jsx]: I have made the changes.

**Comment by Michyjones** [comment]
LGTM

**Pull request merged**